online-shop/src/Products/ProductDetails.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import { IProduct } from '../App';
+import '../Styles/ComponentsStyles/ProductDetails.scss';
+import ConfirmationModal from '../HelperComponents/ConfirmationModal';
+import { Redirect } from 'react-router-dom';
+
+interface IProps {
+  match?: any;
+  addProductToShoppingCartFunction : any;
```

Comment:
This should be typed as function: `addProductToShoppingCartFunction: (product: IProduct) => void`.

Also, you can find a better name for it. E.g. `onAddProduct` or `onAddButtonClick`. Usually callbacks which are linked to events start with `on`.